### DISCUSSION TITLE
Correct way access a customers’s PII

### DISCUSSION THREAD

**Discussion by braeutigamj** [saleweaver/python-amazon-sp-api#207] [FAQ] [answered]
I want to receive Personally Identifiable Information (PII) over `Orders().get_order_address()` but I only receive information like the `PostalCode`. After some search I found the Tokens endpoint and I think I need to create a `create_restricted_data_token()`. So I tried it but it didn't worked well. Can you please check if I made an incorrect call?

My code:
```
       tokens = Tokens(marketplace)
        data_token = tokens.create_restricted_data_token(restrictedResources=[
            {
              'method': 'GET',
              'path': ('/orders/v0/orders/%s/address' %
                       order.external_reference)
            },
            {
              'method': 'GET',
              'path': ('/orders/v0/orders/%s/buyerInfo' %
                       order.external_reference)
            }
        ]).payload['restrictedDataToken']
        orders = Orders(marketplace, refresh_token=data_token)

        payload = orders.get_order_address(order.external_reference).payload
```

My `data_token` looks correct: `'Atz.sprdt|LONGSTRING`
The last call gives me an error: `sp_api.auth.exceptions.AuthorizationError: ('invalid_grant', 'The request has an invalid grant parameter : refresh_token', 400)`

Thanks in advance.

Best Regards
Jonas

**Comment by saleweaver** [comment]
Hi,

did you find a solution to the problem?

**Reply by braeutigamj** [reply]
Hello,
thanks for your response.
I still have this issue. I stay in contact with our "Amazon API Technical Integration Support". The support said that the PII isn't integrated completely by amazon and so far not necessary. She created an internal ticket for the API developer team to fix this issue.

**Reply by saleweaver** [reply]
That's my understanding as well. 
I think the only way to access the data currently is through reports. A representative once answered a support request with:

You may only schedule a report, requesting a report revokes PII access. 

Not sure if this is (still) valid though.

**Reply by braeutigamj** [reply]
That sounds very complicated for that simple requirement... But I will try it. Thanks!

The API developer team will check the problem with the "Shipping API team"...

**Reply by saleweaver** [reply]
Yeah, from experience, just wait a bit and there will be a good solution; the sp-api is still new...

**Reply by braeutigamj** [reply]
Today I received an answer from Amazon. They fixed an issue with the our permissions. After that i only needed to recreate the refresh token by the Amazon Seller Central.

**Comment by braeutigamj** [comment, marked as answer]
Solution: Amazon fixed an issue with our permissions. After that i only needed to recreate the refresh token by the Amazon Seller Central.
The Tokens-endpoint was not necessary.
